python/ql/lib/semmle/python/frameworks/Pycurl.qll

```diff
@@ -58,8 +66,13 @@ module Pycurl {
       override predicate disablesCertificateValidation(
         DataFlow::Node disablingNode, DataFlow::Node argumentOrigin
       ) {
-        // TODO: Look into disabling certificate validation
-        none()
+        exists(API::CallNode c |
+          c = setopt().getACall() and
+          sslverifypeer().getAValueReachableFromSource() = c.getArg(0) and
+          exists(IntegerLiteral i | i.getN() = "0" and c.getArg(1).asExpr() = i)
+        |
+          disablingNode = c and argumentOrigin = c.getArg(1)
+        )
       }
     }
```

Comment:
This class is representing calls to `setopt` where the first argument is specifically `"PyCurl.URL"`. For these calls, the current implementation of `disablesCertificateValidation` as `none()` is correct. What you have done is expressing that this call disables certificate validation if there is _another_ call with arguments `(PyCurl.SSL_VERIFYPEER, 0)`.

What you would need to do is likely to create a new class extending `Http::Client::Request::Range` representing calls to `setopt` where the first argument is specifically `"PyCurl.SSL_VERIFYPEER"` and then you can implement `disablesCertificateValidation` by comparing the second argument with `0` (and also with `False`, please).

Reply:
Closer, but you are still looking at another call (`c = setopt().getACall()`). You have to look at the argument for the call represented by the class; something like `c = this.getACall()`. Also, the boolean has to come from the call; you are now just checking if they wrote `false` anywhere in the code.

Reply:
Yes, much better :-) I think the boolean would also have to go into arg(1)?